dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramIndicator.java

```diff
@@ -148,9 +148,7 @@ public void removeIndicatorGroup( ProgramIndicatorGroup group )
      */
     public Boolean hasNonDefaultBoundaries()
     {
-        return this.analyticsPeriodBoundaries.size() != 2 || ( this.analyticsType == AnalyticsType.EVENT && 
-            !this.analyticsPeriodBoundaries.containsAll( defaultEventTypeBoundaries ) ||
-            this.analyticsType == AnalyticsType.ENROLLMENT );
+        return this.analyticsPeriodBoundaries.size() != 2 ||  this.analyticsType == AnalyticsType.EVENT ;

```

Comment:
As discussed on slack:
add a useIndividualQueries() method that checks both the hasNonDefaultBoundaries() and a new function hasAnalyticsPeriodVariable()

Reply:
fixed